plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/editor/context/project/EncoderServer.kt

```diff
@@ -0,0 +1,260 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cwc.editor.context.project
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.process.KillableProcessHandler
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.project.Project
+import com.intellij.util.io.HttpRequests
+import com.intellij.util.io.createDirectories
+import com.intellij.util.net.NetUtils
+import com.nimbusds.jose.JOSEObjectType
+import com.nimbusds.jose.JWSAlgorithm
+import com.nimbusds.jose.JWSHeader
+import com.nimbusds.jose.crypto.MACSigner
+import com.nimbusds.jwt.JWTClaimsSet
+import com.nimbusds.jwt.SignedJWT
+import org.apache.commons.codec.digest.DigestUtils
+import software.amazon.awssdk.utils.UserHomeDirectoryUtils
+import software.aws.toolkits.core.utils.createParentDirectories
+import software.aws.toolkits.core.utils.exists
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.core.utils.tryDirOp
+import software.aws.toolkits.jetbrains.services.codewhisperer.settings.CodeWhispererSettings
+import software.aws.toolkits.jetbrains.services.cwc.editor.context.project.manifest.ManifestManager
+import java.io.FileOutputStream
+import java.io.IOException
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.attribute.PosixFilePermission
+import java.security.Key
+import java.security.SecureRandom
+import java.util.Base64
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.zip.ZipFile
+import javax.crypto.spec.SecretKeySpec
+
+class EncoderServer(val project: Project) : Disposable {
+    private val cachePath = Paths.get(
+        UserHomeDirectoryUtils.userHomeDirectory()
+    ).resolve(".aws").resolve("amazonq").resolve("cache")
+    val manifestManager = ManifestManager()
+    private val serverDirectoryName = "qserver-${manifestManager.currentVersion}.zip"
```

Comment:
do we need to dynamically resolve the serverDir with getter? not sure if it's possible that manifest version get changed while encodeServer has been initialized

```suggestion
    private val serverDirectoryName: String
         get() = "qserver-${manifestManager.currentVersion}.zip"
```

Reply:
for now the server version is hardcoded in the extension itself which will not change